tests/unit/admin.test.tsx

```diff
@@ -1,12 +1,48 @@
 import '@testing-library/jest-dom'
 import { render, screen } from '@testing-library/react'
 import Page from '../../app/admin/page';
+import { useRouter } from 'next/router';
+import Link from 'next/link';
+
+// Mock next/link component
+jest.mock('next/link', () => {
+    return ({ children, href }) => {
+        return <a href={href}>{children}</a>;
+    };
+});
+
+// Mock next/router
+jest.mock('next/router', () => ({
+    useRouter: jest.fn(),
+}));
 
 describe('Admin', () => {
+    const renderComponent = () => {
+        render(<Page />);
+    };
+
     it('renders the buttons', () => {
         render(<Page />)
         const button = screen.getAllByRole('button')
         const buttonText = screen.getByText('Edit User Role')
         expect(button && buttonText).toBeInTheDocument()
     })
+
+    it('renders the Admin component', () => {
+        renderComponent();
```

Comment:
I see you are creating a render function that is called before each test. Another way to do the same thing is using [beforeEach](https://www.notion.so/Unit-Testing-Guide-0fa6790bf9644c99af39b613f0973006?pvs=4#cb6a7863ac24412a86bedb5c55ded114)